src/bap/bir.py

```diff
@@ -253,7 +253,8 @@ class Values(Map) :
     It is a mapping from the tid of a preceeding block,
     to an expression that denotes a value.
     """
-    pass
+    def __init__(self, *args):
+        super().__init__([ADT(p) for p in args[0]])
```

Comment:
First of all, `super` without args works only in Python 3.x and we are still striving to support both versions. 

Next, I think a more efficient and more robust solution would be to overload the `Map` constructor and just initialize `self.elements`, e.g., something like this:
```python
class Values(Map):
    def __init__(self, *args):
        super(Map,self).__init__(args)
        self.elements = dict(args[0])
```

What do you think?

Reply:
Thanks for your reply! If you want to support both Python versions, I agree.
I was trying to trick Map.__init__. But your way seems neater.